### PR TITLE
remove support for multi-topic messages

### PR DESCRIPTION
**Pull request by blacktemplar**
Removes support for multi topic messages while staying compatible to the old message format. 

If multiple topics are specified in a message only the last one is considered. See also https://github.com/libp2p/go-libp2p-pubsub/pull/388.

### CODE REVIEW

File: protocols/gossipsub/src/mcache.rs [hunk 1]
```diff
@@ -126,7 +126,7 @@ impl<T> MessageCache<T> {
                 let mut found_entries: Vec<MessageId> = entries
                     .iter()
                     .filter_map(|entry| {
-                        if entry.topics.iter().any(|t| t == topic) {
+                        if entry.topic.iter().any(|t| t == topic) {
```
Comment (AgeManning) [inline protocols/gossipsub/src/mcache.rs:129]: This and a few others could just be an `if let Some()...`

File: protocols/gossipsub/src/behaviour.rs [hunk 2]
```diff
@@ -2265,7 +2260,7 @@ where
         let mut recipient_peers = HashSet::new();
 
         // add mesh peers
-        for topic in &message.topics {
+        for topic in &message.topic {
```
Comment (AgeManning) [inline protocols/gossipsub/src/behaviour.rs:2263]: `if let Some()` seems more readable to me

**Review by AgeManning — COMMENTED** [review]
Very minor suggestions

**Comment by AgeManning** [comment]
Although go hasn't done it yet. From the PR it seems they plan to do it in the near future. 

I think it would be good for us to just drop the optional and make a single topic compulsory for each message. I think it might make our code a bit cleaner and we dont have to worry about the case of messages flying around without topics.

**Comment by blacktemplar** [comment]
Ok the latest changes make the topic required. Note that we handle now incoming messages that don't contain a topic as invalid.

This also implicitly resolves your other comments.

**Review by AgeManning — APPROVED** [review]

**Pull request merged**